### PR TITLE
Fix shuffling of test data in infer (#1165)

### LINKED ISSUE #1165: [MembershipInferenceBlackBox] Why does the attack shuffles the inputed data?
Hey, 

I recently used the [MembershipInferenceBlackBox](https://github.com/Trusted-AI/adversarial-robustness-toolbox/blob/main/art/attacks/inference/membership_inference/black_box.py) attack to test my target model. I noticed that if I input data into the `infer()` method, the data points are shuffled before the attack infers the membership status (See: [black_box.py#L273](https://github.com/Trusted-AI/adversarial-robustness-toolbox/blob/24a616befec9d35ba053d16d4a6ac6cfc573039c/art/attacks/inference/membership_inference/black_box.py#L273)). As a result, the returned values can not be matched to their corresponding data points anymore. 

Is this behaviour intended? And if so, it would be great to hear an opinion on why this is intended behaviour.

Thanks!

### PR DESCRIPTION
**Pull request by abigailgold**
Signed-off-by: abigailt <abigailt@il.ibm.com>

# Description

Remove shuffling of test data in infer method

Fixes #1165

## Type of change

Please check all relevant options.

- [ ] Improvement (non-breaking)
- [ X] Bug fix (non-breaking)
- [ ] New feature (non-breaking)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Testing

Black-box membership attack unit tests

**Test Configuration**:
- OS
- Python version
- ART version or commit number
- TensorFlow / Keras / PyTorch / MXNet version

# Checklist

- [X ] My code follows the style guidelines of this project
- [ X] I have performed a self-review of my own code
- [ ] I have commented my code
- [ ] I have made corresponding changes to the documentation
- [ X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by beat-buesser — APPROVED** [review]
Hi @abigailgold Thank you very much for these updates!
